runtime/src/iree/hal/cts/driver_test.h

```diff
@@ -35,8 +35,9 @@ TEST_P(driver_test, QueryAndCreateAvailableDevices) {
                              device_infos[i].name.size)
               << "'";
     iree_hal_device_t* device = NULL;
-    IREE_ASSERT_OK(iree_hal_driver_create_device(
-        driver_, device_infos[i].device_id, iree_allocator_system(), &device));
+    IREE_ASSERT_OK(iree_hal_driver_create_device_by_id(
+        driver_, device_infos[i].device_id, /*param_count=*/0, /*params=*/NULL,
+        iree_allocator_system(), &device));
```

Comment:
Would it be useful to add test coverage in this file for `iree_hal_driver_create_device_by_ordinal`, `iree_hal_driver_create_device_by_path`, and/or `iree_hal_driver_create_device_by_uri`? I think we could at least test that invalid paths / URIs return errors.

Reply:
Ah, now I see 
```
// No path parsing support yet; this is treated like a default device.
```

that's a little concerning... transparently falling back to default device selection could hide errors (asking for a specific type of device should not return a different device)

Reply:
Made all the current implementations check that the path is empty.